### PR TITLE
8345073: Remove SecurityManager checks from sun.awt.image classes

### PR DESCRIPTION
**Pull request by prrace**
Remove SecurityManager related code from sun.awt.image classes

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8345073](https://bugs.openjdk.org/browse/JDK-8345073): Remove SecurityManager checks from sun.awt.image classes (**Bug** - P4)

### Reviewers
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - **Reviewer**) Review applies to [2e56aaf0](https://git.openjdk.org/jdk/pull/22398/files/2e56aaf0582a48c1b404813cec7c8eb19f128308)
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22398/head:pull/22398` \
`$ git checkout pull/22398`

Update a local copy of the PR: \
`$ git checkout pull/22398` \
`$ git pull https://git.openjdk.org/jdk.git pull/22398/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22398`

View PR using the GUI difftool: \
`$ git pr show -t 22398`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22398.diff">https://git.openjdk.org/jdk/pull/22398.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22398#issuecomment-2501833444)
</details>

### CODE REVIEW

**Review by honkar-jdk — APPROVED** [review]
Changes LGTM
Did a 1st pass through the changes and build looks good.

File: src/java.desktop/share/classes/sun/awt/image/URLImageSource.java [hunk 1]
```diff
@@ -106,6 +82,7 @@ protected ImageDecoder getDecoder() {
                 // listed in the original URL, or it can come from one other
                 // host/port that the URL is redirected to.  More than that
                 // and we give up and just throw a SecurityException.
+                // This is not directly related to SecurityManager so keep it.
```
Comment (honkar-jdk) [inline src/java.desktop/share/classes/sun/awt/image/URLImageSource.java:85]: Looks like a valid reason to retain the SecurityException in this case.

File: src/java.desktop/share/classes/sun/awt/image/ImageConsumerQueue.java [hunk 2]
```diff
@@ -68,28 +66,11 @@ static boolean isConsumer(ImageConsumerQueue cqbase, ImageConsumer ic) {
     ImageConsumerQueue(InputStreamImageSource src, ImageConsumer ic) {
         consumer = ic;
         interested = true;
-        // ImageReps do their own security at access time.
-        if (ic instanceof ImageRepresentation) {
-            ImageRepresentation ir = (ImageRepresentation) ic;
-            if (ir.image.source != src) {
-                throw new SecurityException("ImageRep added to wrong image source");
-            }
```
Comment (azvegint) [inline src/java.desktop/share/classes/sun/awt/image/ImageConsumerQueue.java:76]: This exception also doesn't look directly related to the SM, so we may want to keep it, as you did in `URLImageSource#getDecoder`. 

Removing it changed the behavior, now we can instantiate `ImageConsumerQueue` with a mismatched `ImageRepresentation` image source and `InputStreamImageSource`, but we couldn't before.

Comment (prrace) [reply src/java.desktop/share/classes/sun/awt/image/ImageConsumerQueue.java]: > This exception also doesn't look directly related to the SM, so we may want to keep it, as you did in `URLImageSource#getDecoder`.
> 
> Removing it changed the behavior, now we can instantiate `ImageConsumerQueue` with a mismatched `ImageRepresentation` image source and `InputStreamImageSource`, but we couldn't before.

Initially I did leave it, then I convinced myself to take it out. So I'm fine with putting it back. And leaving SE with a comment ..

**Review by azvegint — APPROVED** [review]

**Pull request closed without merging**